homeassistant/components/environment_canada/sensor.py

```diff
@@ -137,42 +111,55 @@ def device_state_attributes(self):
     @property
     def unit_of_measurement(self):
         """Return the units of measurement."""
-        return SENSOR_TYPES[self.sensor_type].get('unit')
+        return self._unit
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     def update(self):
         """Update current conditions."""
         self.ec_data.update()
         self.ec_data.conditions.update(self.ec_data.alerts)
 
+        conditions = self.ec_data.conditions
+        metadata = self.ec_data.metadata
+        sensor_data = conditions.get(self.sensor_type)
+
+        self._unique_id = '-'.join([metadata['location'], self.sensor_type])
         self._attr = {}
+        self._name = sensor_data.get('label')
+        value = sensor_data.get('value')
 
-        sensor_data = self.ec_data.conditions.get(self.sensor_type)
-        if isinstance(sensor_data, list):
+        if isinstance(value, list):
             self._state = ' | '.join([str(s.get('title'))
-                                      for s in sensor_data])
+                                      for s in value])
             self._attr.update({
                 ATTR_DETAIL: ' | '.join([str(s.get('detail'))
-                                         for s in sensor_data]),
+                                         for s in value]),
                 ATTR_TIME: ' | '.join([str(s.get('date'))
-                                       for s in sensor_data])
+                                       for s in value])
             })
         else:
-            self._state = sensor_data
+            self._state = value
+
+        if sensor_data.get('unit') == 'C':
+            self._unit = TEMP_CELSIUS
+        else:
+            self._unit = sensor_data.get('unit')
 
-        timestamp = self.ec_data.conditions.get('timestamp')
+        timestamp = metadata.get('timestamp')
         if timestamp:
             updated_utc = datetime.datetime.strptime(timestamp, '%Y%m%d%H%M%S')
             updated_local = dt.as_local(updated_utc).isoformat()
         else:
             updated_local = None
 
-        hidden = bool(self._state is None or self._state == '')
+        hidden = bool(self._state is None or
```

Comment:
We're phasing out the use of `hidden` attribute. Please remove this.

If we need to mark an entity as temporary unavailable, we should use the entity property `available`.

Reply:
Some of these sensors will be unavailable most of the time, such as the ones for severe weather alerts. Is there a mechanism other than what I've done to automatically show and hide them? I added the `available` property and this causes them to show up as `Unavailable`.

Reply:
The user can configure showing entities depending on state in lovelace. I suggest just removing hidden and let them return unknown state (`None`) or similar, ie not implement available either.

Reply:
Done